clerk.go

```diff
@@ -267,19 +286,32 @@ func (b *defaultBackend) do(req *http.Request, params Params, setter ResponseRea
 	return nil
 }
 
-// Sets the params in either the request body, or the querystring
-// for GET requests.
-func setRequestBody(req *http.Request, params Params) error {
+// Sets the APIRequest params in either the request body, or the
+// querystring for GET requests.
+// If the APIRequest is multipart, the http.Request Content-Type
+// is overwritten and the body will be sent as a multipart message.
+func setRequestBody(req *http.Request, apiReq *APIRequest) error {
 	// GET requests don't have a body, but we will pass the params
 	// in the query string.
 	if req.Method == http.MethodGet {
-		setRequestQuery(req, params)
+		setRequestQuery(req, apiReq.Params)
 		return nil
 	}
 
-	body, err := json.Marshal(params)
-	if err != nil {
-		return err
+	var body []byte
+	var err error
+	if apiReq.isMultipart {
+		var contentType string
+		body, contentType, err = apiReq.Params.ToMultipart()
+		if err != nil {
+			return err
+		}
+		req.Header.Set("Content-Type", contentType)
```

Comment:
❓ Is the content type dynamic here or is it a specific value always?

Reply:
The content-type is multipart/form-data but also contains the writer's boundary.

https://cs.opensource.google/go/go/+/refs/tags/go1.22.0:src/mime/multipart/writer.go;l=74

I believe this is how the server knows how to parse the body successfully and figure out where each part ends.

https://datatracker.ietf.org/doc/html/rfc7578#section-4.1

Reply:
TIL, thanks for explaining!